rust-connectors/sinks/sql/src/transform.rs

```diff
@@ -31,131 +32,22 @@ impl Transformations {
                 param.insert(PARAM_WITH.to_string(), with);
             }
             let raw = downloader.download_binary(step.uses.as_str()).await?;
-            let payload = LegacySmartModulePayload {
-                wasm: SmartModuleWasmCompressed::Raw(raw),
-                kind: SmartModuleKind::Map,
-                params: param.into(),
-            };
-            let engine = SmartEngine::default();
-            let mut smart_module = engine.create_module_from_payload(payload, None)?;
-            smart_module.invoke_constructor()?;
-            smart_modules.push(smart_module);
-        }
-        Ok(Self { smart_modules })
-    }
+            let mut sm_builder = SmartModuleConfig::builder();
+            let sm_builder = sm_builder.params(param.into());
 
-    pub fn transform(&mut self, input: Record) -> anyhow::Result<Vec<Record>> {
-        let mut result = vec![input];
-        for smart_module in self.smart_modules.iter_mut() {
-            let input = SmartModuleInput::try_from(result)?;
-            let output = smart_module.process(input)?;
-            if let Some(err) = output.error {
-                return Err(err.into());
-            }
-            result = output.successes;
+            builder.add_smart_module(sm_builder.build()?, raw)?;
         }
-        Ok(result)
-    }
 
-    pub fn size(&self) -> usize {
-        self.smart_modules.len()
```

Comment:
Maybe we should implement num_instances() in `SmartModuleChainInstance`

Reply:
I suppose that's not a bad idea. I just don't think it's all that helpful to know the number of smartmodules that's why I deleted it.